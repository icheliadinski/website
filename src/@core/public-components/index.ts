export { About } from './about'
export { Contact } from './contact'
export { Blog } from './blog'
export { Skills } from './skills'
export { Works } from './works'
