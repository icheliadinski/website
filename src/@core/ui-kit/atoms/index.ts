export { Loader } from './loader'
export { ICON_TYPES, Icon } from './icon'
