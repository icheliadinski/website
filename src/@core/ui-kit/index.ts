export { Loader, Icon, ICON_TYPES } from './atoms'

export { LeftMenu } from './molecules'
