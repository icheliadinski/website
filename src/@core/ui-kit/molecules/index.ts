export { LeftMenu } from './left-menu'
