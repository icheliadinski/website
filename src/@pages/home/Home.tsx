import React from 'react'
import { Switch, Route } from 'react-router-dom'
import { LeftMenu, ICON_TYPES } from '@core/ui-kit'
import { About, Contact, Skills, Works, Blog } from '@core/public-components'

export const Home = () => {
  const navigation = [
    { path: '', icon: ICON_TYPES.USER_ICON },
    { path: 'contact', icon: ICON_TYPES.PHONE_ICON },
    { path: 'blog', icon: ICON_TYPES.EDIT_ICON },
    { path: 'works', icon: ICON_TYPES.MONITOR_ICON },
    { path: 'skills', icon: ICON_TYPES.GLASSES_ICON },
  ]

  return (
    <main>
      <LeftMenu navigationItems={navigation} />
      <Switch>
        <Route path="/home" component={About} exact />
        <Route path="/home/contact" component={Contact} />
        <Route path="/home/blog" component={Blog} />
        <Route path="/home/works" component={Works} />
        <Route path="/home/skills" component={Skills} />
      </Switch>
    </main>
  )
}
