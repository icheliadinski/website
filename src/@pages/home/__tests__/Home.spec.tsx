import React from 'react'
import { shallow, ShallowWrapper } from 'enzyme'
import { ICON_TYPES } from '@core/ui-kit'
import { Home } from '../Home'

type OnLeftItemClickFn = (item: ICON_TYPES) => void

describe('[Home]', () => {
  let wrapper: ShallowWrapper

  beforeEach(() => {
    wrapper = shallow(<Home />)
  })

  it('renders successfully', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
