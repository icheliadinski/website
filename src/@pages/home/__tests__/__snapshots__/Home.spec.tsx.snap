// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Home] renders successfully 1`] = `
<main>
  <LeftMenu
    activeItem="user"
    onItemClick={[Function]}
    profileIcons={
      Array [
        "user",
        "phone",
        "edit",
        "monitor",
        "glasses",
      ]
    }
  />
</main>
`;
